### DISCUSSION TITLE
[WASM] Using Microsoft.Graphics.Canvas.UI.Composition

### DISCUSSION THREAD

**Discussion by SanAlexis** [unoplatform/uno#11613] [Q&A] [answered]
I'm doing a migration project from a project targeting UWP `uap10.0.17134.0` to a project targeting WASM `net6.0`. I have CanvasDevice class which comes from `Microsoft.Graphics.Canvas.UI.Composition` and I'm unable to embed any of these dependencies to make it work:
* [Win2D.uwp](https://www.nuget.org/packages/Win2D.uwp/1.27.0-preview2)
* [Microsoft.Graphics.Win2D](https://www.nuget.org/packages/Microsoft.Graphics.Win2D)
* [Win2D.NET](https://www.nuget.org/packages/Win2D.NET)
* [SkiaSharp](https://www.nuget.org/packages/SkiaSharp)

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the question. Win2D is not supported by uno targets (iOS, Android, GTK/WPF and Wasm) though you can use it in a [platform specific way](https://platform.uno/docs/articles/platform-specific-csharp.html) on Windows targets (`.UWP` and `.Windows` heads).

SkiaSharp is supported on all targets, see [this sample](https://github.com/unoplatform/Uno.Samples/tree/master/UI/SkiaSharpTest).
